lib/python/pathling/spark.py

```diff
@@ -0,0 +1,89 @@
+#  Copyright 2025 Commonwealth Scientific and Industrial Research
+#  Organisation (CSIRO) ABN 41 687 119 230.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+import uuid
+
+from py4j.java_gateway import JavaObject, JVMView
+from pyspark import SparkContext
+from pyspark.sql import SparkSession
+
+
+class Dfs:
+    """A class for interacting with the Hadoop Distributed File System (HDFS) in Spark."""
+
+    def __init__(self, spark: SparkSession):
+        """
+        Initialize the Dfs class with a SparkSession.
+
+        :param spark: SparkSession instance
+        """
+        if not spark:
+            raise ValueError("SparkSession must be provided")
+        sc: SparkContext = spark.sparkContext
+        self._jvm: JVMView = sc._jvm
+        self._hadoop_conf: JavaObject = sc._jsc.hadoopConfiguration()
+        self._fs = self._jvm.org.apache.hadoop.fs.FileSystem.get(self._hadoop_conf)
+
+    def get_temp_dir_path(self, infix: str = "app-tmp",
+                          def_temp_base: str = "/tmp", qualified=True) -> str:
+        """
+        Returns a unique path for a temporary directory in Spark's filesystem.
+    
+        The path is constructed by appending a UUID to the base temporary directory, 
+        ensuring uniqueness for each call.
+        The directory itself is not created, only the path is returned.
+        
+        :param infix: String to insert between the base directory and the UUID (default: "app-tmp").
+        :param def_temp_base: Fallback base directory if `hadoop.tmp.dir` is not set in Hadoop configuration (default: "/tmp").
+        :param qualified: If True, returns a fully qualified Hadoop path; if False, returns a raw path string.
+        :return: String representing the unique temporary directory path.
+        """
+        base_tmp = self._hadoop_conf.get("hadoop.tmp.dir") or def_temp_base
+        uuid_suffix = str(uuid.uuid4())
+        tmp_path = f"{base_tmp}/{infix}-{uuid_suffix}"
```

Comment:
Will this work in Windows?

Reply:
It should/might is spark/hadoop is configured corectly becasue it uses hadoop's Path() class which even on windows uses forward slash ('/') as separator and the 'hadoop.tmp.dir' should be set to something valid on windows.

Reply:
Yes, pathling-windows.it.csiro.au